src/Command/Ingester/CreateUserCommand.php

```diff
@@ -0,0 +1,461 @@
+    }
+
+    private function getFindLineItemCriteria(): FindLineItemCriteria
+    {
+        $criteria = new FindLineItemCriteria();
+
+        if (!empty($this->lineItemIds)) {
+            $criteria->addLineItemIds(...$this->lineItemIds);
+        }
+        return $criteria;
+    }
+
+    /**
+     * @throws LineItemNotFoundException
+     */
+    private function getAllLineItemSlugs(): Void {
+        $lineItems = $this->lineItemRepository->findAllSlugsAsArray();
+        if (empty($lineItems)) {
+            throw new LineItemNotFoundException("No line items were found in database.");
+        }
+        $this->lineItemSlugs = array_column($lineItems, 'slug');
+    }
+
+    /**
+     * Function to check whether a user exist for a Line Item Slug and returns last increment no of it
+     * 
+     */
+    private function checkLineItemSlugUserExist(
+        LineItemCollection $lineItems,
+        Array $lineItemSlugs
+    ): Array {
+        $userNameIncArr = [];
+        foreach ($lineItemSlugs as $slug) {
+            $lineItem = $lineItems->checkSlugExistOrNot($slug);
+            if(!empty($lineItem)) {
+                $lineItemId = (int)$lineItem->getId();
+                $assignment = $this->assignmentRepository->findByLineItemId($lineItemId);
+                if (!empty($assignment)) {
+                    $userInfo = $assignment->getUser()->getUsername();
+                    $userNameArr = explode('_', $userInfo);
+                    $userNameLastNo = preg_match("/^\d+$/", end($userNameArr)) ? (int)end($userNameArr) : self::DEFAULT_USERNAME_INCREMENT_NO;
+                    $userNameIncArr[$slug] = $userNameLastNo;
+                } else {
+                    $userNameIncArr[$slug] = self::DEFAULT_USERNAME_INCREMENT_NO;
+                }
+            } else {
+                $this->symfonyStyle->note($slug. " LineItem Slug not exist in the system");
+                $this->lineItemSlugs = array_diff( $this->lineItemSlugs, [$slug] );
+            }
+        }
+        return $userNameIncArr;
+    }
+
+    /**
+     * Function to create new groupIds based on LTI Instance count 
+     */
+    public function getLoadBalanceGroupID(string $groupPrefix): Array
+    {
+        $totalInstance = $this->ltiInstanceRepository->findAllAsCollection()->count();
+        $targetId = 1;
+        $groupIds = [];
+        while ($targetId <= $totalInstance) {
+            $random = substr(md5(random_bytes(10)), 0, 10);
+            $groupId = sprintf($groupPrefix.'_%s', $random);
+            array_push($groupIds, $groupId);
+            $targetId++;
+        }
+        return $groupIds;
+    }
+
+    /**
+     * Function to create a new csv and save data to it
+     */
+    private function writeCsvData(string $path, array $head, array $data): Void
+    {
+        $csv = Writer::createFromPath($path, "w");
+        $csv->insertOne($head);
+        $csv->insertAll($data);
+    }
```

Comment:
Would expect a separated class for handling the CSVs writing

Reply:
Done